server/src/main/java/com/saralein/server/response/Header.java

```diff
@@ -1,14 +1,19 @@
 package com.saralein.server.response;
 
-import static com.saralein.server.Constants.CRLF;
 import com.saralein.server.protocol.StatusCodes;
+import com.saralein.server.request.transfer.Cookie;
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
+import static com.saralein.server.Constants.CRLF;
 
 public class Header {
```

Comment:
I feel like we might need to reevaluate this class. It could be just the name, but I have a suspicion it might not be that simple.

I think the idea here is that the `Header` class represents the _message header_, which contains _header fields_, yeah? 

That can be a little confusing as someone might interpret this as being a class that represents a single header field from the name. Usually, in terms of HTTP, we think of a header as being a single header field.

I wonder if we are better off splitting this class into multiple classes. `StatusLine`, `HeaderFields`, `Header`. Then you could perhaps have a `Message` class that pulls all of the parts of the HTTP message and presents them as a string or a byte array. 

It doesn't need to be that exact implementation, that is just a potential avenue, but I do think we need to extract some responsibilities here and do some renaming to match the language of HTTP.

Reply:
I think @zacharyjdavy started hinting at this in another PR recently.

Reply:
Will take a look.  The header has felt off for a while and more so when I started changing the parser and I could almost reuse pieces of what I'm currently using in the response for the request, but not quite because it had that status line in there.